installer/pkg/common/ca.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package common
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	cacerts      = "cacerts"
+	etcSSLCerts  = "/etc/ssl/certs"
+	caVolumeName = "gitpod-ca-certificate"
+	caCrtKey     = "ca.crt"
+
+	updateContainerdScript = `TARGETS="docker containerd";for TARGET in $TARGETS;do mkdir -p /etc/$TARGET/certs.d/reg.%s:%v && echo "$GITPOD_CA_CERT" > /etc/$TARGET/certs.d/reg.%s:%v/ca.crt && echo "OK";done`
+)
+
+func InternalCAVolumeMount() *corev1.VolumeMount {
+	return &corev1.VolumeMount{
+		Name:      cacerts,
+		ReadOnly:  true,
+		MountPath: etcSSLCerts,
+	}
+}
+
+func InternalCAVolume() *corev1.Volume {
+	return &corev1.Volume{
+		Name: caVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+}
+
+// InternalCAContainer returns a container that updates the CA certificates
+// adding the internal Gitpod self-generated CA.
+// This is required for compoenents that communitate with registry-facade
```

Comment:
Couple of typos in the comment:

`compoenents` => "components"
`communitate` => "communicate"
`ontainerd` => "containerd"

Reply:
fixed